apps/connect/source/ftrack_connect/ui/application.py

```diff
@@ -670,72 +696,78 @@ def configureConnectAndDiscoverPlugins(self):
 
         self._discover_connect_widgets()
 
-    def _discover_plugin_paths(self):
-        '''Return a list of paths to pass to ftrack_api.Session()'''
+    def _gather_plugins(self, plugin_directory):
+        '''Return plugin data from *plugin_directory*.'''
 
-        plugin_paths = []
+        result = []
+        if not plugin_directory:
+            return result
+        self.logger.debug(
+            u'Searching {0!r} for plugin hooks.'.format(plugin_directory)
+        )
+        if os.path.isdir(plugin_directory):
+            for candidate in get_plugins_from_path(plugin_directory):
+                candidate_path = os.path.join(plugin_directory, candidate)
 
-        plugin_paths.extend(self._gather_plugins(self.defaultPluginDirectory))
+                # Check for hook folder
+                full_hook_path = os.path.join(candidate_path, 'hook')
+                if not os.path.isdir(full_hook_path):
+                    self.logger.debug(
+                        f'Missing "hook" folder: {candidate_path}'
+                    )
+                    continue
```

Comment:
This is already checked in the incompatible plugin, is it needed to re-check in here?